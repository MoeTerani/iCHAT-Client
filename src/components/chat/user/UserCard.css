/*--------------------
status- Title
--------------------*/
.userCard--title {
  display: flex;
  justify-content: center;
  align-items: center;
  -webkit-box-flex: 0;
  /* flex: 0 1 45px; */
  position: relative;
  z-index: 2;
  background: rgba(0, 0, 0, 0);
  color: #fff;
  text-transform: uppercase;
  text-align: left;
  padding: 10px 10px 10px 50px;
}
.userCard--title h1,
.userCard--title h2 {
  font-weight: normal;
  font-size: 1rem;
  margin: 0;
  padding: 0;
}
.userCard--title h2 {
  color: rgba(255, 255, 255, 0.5);
  font-size: 8px;
  letter-spacing: 1px;
}
.userCard--title .avatar {
  position: absolute;
  z-index: 1;
  top: 8px;
  left: 9px;
  border-radius: 30px;
  width: 30px;
  height: 30px;
  overflow: hidden;
  margin: 0;
  padding: 0;
  border: 2px solid rgba(255, 255, 255, 0.24);
}
.userCard--title .avatar img {
  width: 100%;
  height: auto;
}
