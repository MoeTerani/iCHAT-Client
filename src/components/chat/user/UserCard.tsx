import React from 'react';
import './UserCard.css';

interface Props {}

const UserCard = ({ name, avatar }: { name: string; avatar: string }) => {
  return (
    <div className='userCard--title'>
      <h1>{name} </h1>
      <div className='avatar'>
        <img src={avatar} alt='current user avatar' />
      </div>
    </div>
  );
};

export default UserCard;
