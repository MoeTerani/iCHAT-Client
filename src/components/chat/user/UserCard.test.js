import { shallow } from 'enzyme';
import React from 'react';
import UserCard from './UserCard';
import {
  getWrapperShallow,
  getWrapperComponent,
} from '../../../utilities/testHelperFn';

describe('UserCard components', () => {
  let props;
  let Component;

  beforeEach(() => {
    props = {
      name: '',
      avatar: '',
    };
    Component = getWrapperShallow(UserCard, { ...props });
  });
  test('it renders without errors', () => {
    expect(Component).toMatchSnapshot();
  });
  test('should render without errors', () => {
    expect(Component.length).toEqual(1);
  });
});
