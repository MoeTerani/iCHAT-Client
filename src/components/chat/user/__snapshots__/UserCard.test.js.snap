// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard components it renders without errors 1`] = `ShallowWrapper {}`;
